pkg/cluster/cluster.go

```diff
@@ -409,6 +409,12 @@ func (c *Cluster) compareStatefulSetWith(statefulSet *appsv1.StatefulSet) *compa
 		reasons = append(reasons, "new statefulset's pod management policy do not match")
 	}
 
+	if !reflect.DeepEqual(c.Statefulset.Spec.PersistentVolumeClaimRetentionPolicy, statefulSet.Spec.PersistentVolumeClaimRetentionPolicy) {
```

Comment:
we have a failing e2e test - so somehow this comparison must lead to a statefulset replacement. Can the order of the two fields be swapped? Maybe we need an explicit comparison method for the retention policy. Or some side is simply not set and nil.